src/lib.rs

```diff
@@ -202,9 +207,45 @@ impl Builder {
         self
     }
 
+    /// Return `Logger` and TLS connection background task.
+    #[cfg(feature = "tokio-rustls")]
+    pub fn connect_tls<T>(
+        self,
+        addr: T,
+        domain_name: String,
```

Comment:
We should take `domain_name` as a `&str` rather than `String` if it's only use is as an argument to `pub fn try_from_ascii_str(dns_name: &str)`